Assets/Scripts/Player/PlayerProgress.cs

```diff
@@ -10,6 +10,7 @@ public class PlayerProgress : MonoBehaviour
     [Header("Debugging properties")]
     [SerializeField]
     public Checkpoint currentCheckpoint;
+    private int deaths;

```

Comment:
you could use a property to prevent needing a getter:
`public int Deaths { get; private set}`

Reply:
Oh, that's some nice syntactic sugar. Didn't know about that.